pinot-server/src/main/java/com/linkedin/pinot/server/api/resources/MmapDebugResource.java

```diff
@@ -0,0 +1,71 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.server.api.resources;
+
+import com.linkedin.pinot.common.utils.MmapUtils;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import org.apache.commons.lang3.tuple.Pair;
+import org.restlet.resource.ResourceException;
+
+
+/**
+ * Debug endpoint to check memory allocation.
+ */
+@Api(value="debug", description="Debug information", tags = "Debug")
+@Path("debug")
+public class MmapDebugResource {
+
+  public static class AllocationInfo {
+    String context;
+    String type;
+    int size;
```

Comment:
I suppose this shows up as 'size' in the output? Would it be better to use sizeInBytes (or as appropriate)?